app/src/main/java/com/duckduckgo/app/browser/BrowserTabFragment.kt

```diff
@@ -3701,7 +3726,9 @@ class BrowserTabFragment :
                 }
 
                 if (shouldUpdateOmnibarTextInput(viewState, viewState.omnibarText)) {
-                    omnibar.omnibarTextInput.setText(viewState.omnibarText)
+                    if (!viewState.navigationChange) {
```

Comment:
is this right? you have a method that checks if the text in the omnibar should change, but then you override the behaviour here. which one is it?